### PR TITLE
[CSM] Hook ftruncate

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR hooks the `ftruncate` syscall.

### Motivation

This syscall was forgotten in the list of hooked syscalls. We currently support `truncate` but forgot `ftruncate`.

### Describe how to test/QA your changes

A functional test was added.

### CODE REVIEW

**Review by spikat — APPROVED** [review]

File: pkg/security/ebpf/probes/event_types.go [hunk 1]
```diff
@@ -149,10 +149,12 @@ func GetSelectorsPerEventType(fentry bool) map[eval.EventType][]manager.ProbesSe
 			// Open probes
 			&manager.AllOf{Selectors: []manager.ProbesSelector{
```
Comment (YoannGh) [inline pkg/security/ebpf/probes/event_types.go:150]: Shouldn't we use `OneOf` instead of `AllOf` here?

Comment (Gui774ume) [reply pkg/security/ebpf/probes/event_types.go:150]: Not yet :( I first need to find a stable hook point or a combination of hook points that will cover the ftruncate syscall.

Comment (Gui774ume) [reply pkg/security/ebpf/probes/event_types.go:150]: Well idk if you intuited the answer or if you looked it up, but you ended up being almost right 😅 
For some reason, they introduced `security_file_truncate` in 6.2, and replaced `security_path_truncate` with it only for `do_sys_ftruncate`. So in the end, changing it to "besteffort" should do the trick ! ("oneof" would prevent hooking both of them when they're available - which we need to support ftruncate on recent kernels)

Comment (paulcacheux) [reply pkg/security/ebpf/probes/event_types.go:150]: Are you sure OneOf prevents the hooking of both ? My understanding was that it was kind of "at least one" semantics

**Pull request closed without merging**